File: trilobytelib/src/main/java/com/st/trilobyte/models/Property.java
package com.st.trilobyte.models;

import java.io.Serializable;

public class Property<T> implements Serializable {

    public enum PropertyType {
        FLOAT, INT, ENUM, STRING, BOOL
    }

    private String label;

    private PropertyType type;

    private T value;

    public String getLabel() {
        return label;
    }

    public void setLabel(final String label) {
        this.label = label;
    }

    public PropertyType getType() {
        return type;
    }

    public void setType(final PropertyType type) {
        this.type = type;
    }

    public T getValue() {
        return value;
    }

    public void setValue(final T value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "Property{" +
                "label='" + label + '\'' +
                ", type=" + type +
                ", value=" + value +
                '}';
    }
}

File: st_flow_demo/src/main/java/com/st/flow_demo/models/Property.kt
package com.st.flow_demo.models

import com.st.flow_demo.serializer.PropertySerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable (with =  PropertySerializer::class)
data class Property(
    @SerialName(value = "label")
    var label: String,
    @SerialName(value = "type")
    var type: PropertyType,
    @SerialName(value = "value")
    var value: Any,
    @SerialName(value = "enumValues")
    var enumValues: List<PropertyEnumValue>?=null) {
    override fun toString(): String {
        return "Property{" +
                "label='" + label + '\'' +
                ", type=" + type +
                ", value=" + value +
                '}'
    }
}

enum class PropertyType {
    FLOAT, INT, ENUM, STRING, BOOL
}

